src/Renderer/Shader/TileFS.glsl

```diff
@@ -122,21 +122,25 @@ void main() {
                             textureIndex,
                             projWGS84 ? vUv_WGS84 : uvPM);
 
-                        if (layerColor.a > 0.0) {
+                        bool isNotTransparent = layerColor.a > 0.0;
+
+                        // if first texture is transparent, we need to force blending with CWhite
+                        if (isNotTransparent || (!isNotTransparent && !validTexture)) {
                             validTexture = true;
-                            float lum = 1.0;
-
-                            if(paramsA.z > 0.0) {
-                                float a = max(0.05,1.0 - length(layerColor.xyz-CWhite.xyz));
-                                if(paramsA.z > 2.0) {
-                                    a = (layerColor.r + layerColor.g + layerColor.b)*0.333333333;
-                                    layerColor*= layerColor*layerColor;
-                                }
-                                lum = 1.0-pow(abs(a),paramsA.z);
+                            // unassociate the alpha (because id premultiply alpha texture)
+                            if (isNotTransparent) {
```

Comment:
What's the point of hacking the premultiplied alpha flow instead of using it completely (by using material.premultipliedAlpha)?

(at the very least you must add a comment explaining why you don't make this obvious choice)

Reply:
We use premultiplied alpha for mixing color layers
We don't use material.premultipliedAlpha which is used to blend the materials

Reply:
You still didn't answer my question: why don't you use `material.premultipliedAlpha`?

Reply:
We don't use `material.premultipliedAlpha` which is used to blend materials/mesh in rendering buffer with `blendFunc`.
Here the blending takes place in the shader, and we can't do it with the webgl functions (blendFunc...)

Reply:
`̀// We use premultiplied alpha for mixing color layers` is wrong since you demultiply the color before the blending. 
Premultiply-alpha is only used for texture fetching here and avoid gl_linear filtering artifact.

Regarding `material.premultipliedAlpha` let's not use it for now.

Reply:
yes, I change the comment